framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/ImmutableKieServerScenarioImpl.java

```diff
@@ -0,0 +1,136 @@
+import org.kie.cloud.openshift.operator.model.KieApp;
+import org.kie.cloud.openshift.operator.model.components.Auth;
+import org.kie.cloud.openshift.operator.model.components.Server;
+import org.kie.cloud.openshift.operator.model.components.Sso;
+import org.kie.cloud.openshift.util.SsoDeployer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ImmutableKieServerScenarioImpl extends OpenShiftOperatorScenario<ImmutableKieServerScenario> implements ImmutableKieServerScenario {
+
+    private KieServerDeploymentImpl kieServerDeployment;
+    private boolean deploySso;
+    private SsoDeployment ssoDeployment;
+
+    private static final Logger logger = LoggerFactory.getLogger(ImmutableKieServerScenarioImpl.class);
+
+    public ImmutableKieServerScenarioImpl(KieApp kieApp, boolean deploySso) {
+        super(kieApp);
+        this.deploySso = deploySso;
+    }
+
+    @Override
+    protected void deployCustomResource() {
+        if (deploySso) {
+            ssoDeployment = SsoDeployer.deploySecure(project);
+            URL ssoSecureUrl = ssoDeployment.getSecureUrl().orElseThrow(() -> new RuntimeException("RH SSO secure URL not found."));
+
+            Sso sso = new Sso();
+            sso.setAdminUser(DeploymentConstants.getSsoServiceUser());
+            sso.setAdminPassword(DeploymentConstants.getSsoServicePassword());
+            sso.setUrl(SsoDeployer.createSsoEnvVariable(ssoSecureUrl.toString()));
+            sso.setRealm(DeploymentConstants.getSsoRealm());
+            sso.setDisableSSLCertValidation(true);
+
+            Auth auth = new Auth();
+            auth.setSso(sso);
+            kieApp.getSpec().setAuth(auth);
+        }
+
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            registerCustomTrustedSecret(server);
+        }
+
+        // deploy application
+        getKieAppClient().create(kieApp);
+
+        kieServerDeployment = new KieServerOperatorDeployment(project, getKieAppClient());
+        kieServerDeployment.setUsername(DeploymentConstants.getKieServerUser());
+        kieServerDeployment.setPassword(DeploymentConstants.getKieServerPassword());
+
+        logger.info("Waiting until all services are created.");
+        try {
+            new SimpleWaiter(() -> kieServerDeployment.isReady()).reason("Waiting for Kie server service to be created.").timeout(TimeUnit.MINUTES, 1).waitFor();
+        } catch (WaiterException e) {
+            throw new RuntimeException("Timeout while deploying application.", e);
+        }
+
+        logger.info("Waiting for Kie server deployment to become ready.");
+        kieServerDeployment.waitForScale();
+
+        logNodeNameOfAllInstances();
+    }
+
+    @Override
+    public KieServerDeployment getKieServerDeployment() {
+        return kieServerDeployment;
+    }
+
+    @Override
+    public List<Deployment> getDeployments() {
+        List<Deployment> deployments = new ArrayList<>(Arrays.asList(kieServerDeployment));
```

Comment:
Add missing SSO Deployment
```suggestion
        List<Deployment> deployments = new ArrayList<>(Arrays.asList(kieServerDeployment,ssoDeployment));
```